### PR TITLE
Added support for Suffix Providers

### PR DESCRIPTION
**Pull request by icrc-loliveira**
Similar to Prefix providers with this development is possible to provide more complex suffixs for the identifiers.

### CODE REVIEW

**Comment by icrc-loliveira** [comment]
Hello @mseaton 
Can please review or assign someone to review this PBI?

cc: @icrc-fdeniger

File: api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java [hunk 1]
```diff
@@ -79,8 +89,14 @@ public String getIdentifierForSeed(long seed) {
 			prefixProvider = new ConstantPrefixProvider(prefix);
 		}
 		identifier = prefixProvider.getValue() + identifier;
-    	identifier = (suffix == null ? identifier : identifier + suffix);
-    	
+
+
+		SuffixProvider sufixProvider = getSuffixProvider(suffix);
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java:94]: Misspelling of variable name

**Review by mseaton — COMMENTED** [review]
Generally looks fine to me.  It seems less than ideal to have to create duplicate interfaces and classes for Suffix Providers when they already exist for Prefix Providers.  Thoughts on whether we can avoid this somehow?  Or is the naming such that it is better just to duplicate everything?

**Comment by icrc-loliveira** [comment]
Hello @mseaton,
Thanks for your review.  
It should be possible to do this implementation without to many duplication. However for it to make sense will most likely impact other distributions that are already using the Prefix Providers.

**Comment by icrc-loliveira** [comment]
Hello,
Can you please finish reviewing this PR.

cc: @samuelmale, @mseaton, @ibacher

**Review by mseaton — APPROVED** [review]
I don't love a lot of the way this was implemented, but most of that is with the existing code that this is replicating, so I'm fine doing that.

**Pull request merged**